src/displacements.py

```diff
@@ -0,0 +1,225 @@
+    offsets = calculate_cell_offsets_from_coords(traj_coords)
+
+    corrected_coords = traj_coords + offsets
+
+    displacements = []
+
+    first = corrected_coords[equilibration_steps]
+
+    for disp in corrected_coords[equilibration_steps:]:
+        diff_vectors = disp - first
+        lengths = calculate_lengths(diff_vectors,
+                                    metric_tensor=lattice.metric_tensor)
+        displacements.append(lengths)
+
+    displacements = np.array(displacements)
+
+    return displacements
+
+
+def plot_displacement_per_site(displacements: np.ndarray):
+    """Plot displacement per site.
+
+    Parameters
+    ----------
+    displacements : np.ndarray
+        Numpy array with displacements
+    """
+    fig, ax = plt.subplots()
+
+    for site_displacement in displacements.T:
+        ax.plot(site_displacement, lw=0.3)
+
+    ax.set(title='Displacement of diffusing element',
+           xlabel='Time step',
+           ylabel='Displacement (Angstrom)')
+
+    plt.show()
+
+
+def plot_displacement_per_element(species: list[Element],
+                                  displacements: np.ndarray):
+    """Plot displacement per element.
+
+    Parameters
+    ----------
+    structure : Structure
+        Pymatgen structure used for labelling
+    displacements : np.ndarray
+        Numpy array with displacements
+    """
+    from collections import defaultdict
+
+    grouped = defaultdict(list)
+
+    for specie, displacement in zip(species, displacements.T):
+        grouped[specie.name].append(displacement)
+
+    fig, ax = plt.subplots()
+
+    for specie, displacement in grouped.items():
+        mean_disp = np.mean(displacement, axis=0)
+        ax.plot(mean_disp, lw=0.3, label=specie)
+
+    ax.legend()
+    ax.set(title='Displacement per element',
+           xlabel='Time step',
+           ylabel='Displacement (Angstrom)')
+
+    plt.show()
+
+
+def plot_displacement_histogram(displacements: np.ndarray):
+    """Plot histogram of total displacement at final timestep.
+
+    Parameters
+    ----------
+    displacements : np.ndarray
+        Numpy array with displacements
+    """
+    fig, ax = plt.subplots()
+    ax.hist(displacements[-1])
+    ax.set(title='Histogram of displacement of diffusing element',
+           xlabel='Displacement (Angstrom)',
+           ylabel='Nr. of atoms')
+    plt.show()
+
+
+if __name__ == '__main__':
+    from pathlib import Path
+
+    import yaml
+    from pymatgen.io.vasp.outputs import Vasprun
+
+    vasp_xml = '/run/media/stef/Scratch/md-analysis-matlab-example/vasprun.xml'
+
+    path_coords = Path('traj_coords.npy')
+    path_data = Path('data.yaml')
+
+    # skip first timesteps
+    equilibration_steps = 1250
+
+    if not (path_coords.exists() and path_data.exists()):
+        vasprun = Vasprun(
+            vasp_xml,
+            parse_dos=False,
+            parse_eigen=False,
+            parse_projected_eigen=False,
+            parse_potcar_file=False,
+        )
```

Comment:
Fine for now, but I think we should do the parsing (and configuration) in a separate file (much like `analyse_md` did). then we can still add a `__main__` that calls the plot through this separate file.
In this way it becomes easier to only parse the data once for an arbitrary number of plots. (even if its cached)

Reply:
But lets fix this later, when we add more plots